EngineLayer/Proteomics/CompactPeptide.cs

```diff
@@ -30,6 +32,33 @@ public CompactPeptide(PeptideWithSetModifications peptideWithSetModifications)
             MonoisotopicMassIncludingFixedMods = peptideWithSetModifications.MonoisotopicMass;
         }
 
+        public CompactPeptide(PeptideWithSetModifications peptideWithSetModifications, TerminusType terminusType)
+        {
+            double theMass = 0;
+            NTerminalMasses = new double[1];
```

Comment:
Oh boy am I finding a lot of logic errors as I try to convert this thing to null. Gonna need a few minutes.